ruby/sorting-words-and-numbers/main.rb

```diff
@@ -1,16 +1,19 @@
 # frozen_string_literal: true
+SEPARATER = ' '
 
-loop do
-  sorted = [[], []]
-  line = gets
-  next if line.nil?
+while line = gets.chomp
+  classified_words = {
+    numerical_digits: [],
+    others: [],
+  }
 
-  line.split(' ').each do |word|
-    if /\d+/.match(word)
-      sorted[1].append(word)
+  line.split(SEPARATER).each do |word|
+    if word.match?(/^\d+$/)
```

Comment:
[BUGFIX] Add ^ and $ to regular expression for condition.

Reply:
`Regexp#match` is a method that returns the result as a `MatchData` object when a match occurs, not a method used for conditions.

Also, the subject is not Regexp, but `word`.

Therefore,  I changed the method to `String#match?`